README.md

```diff
@@ -196,10 +212,245 @@ The following syntax shall be used to declare an array:
 ```YAML
 # Array of datatype uint32, by default size of the array is undefined
 datatype: uint32[]
+
 # Optional: specified number of elements in the array
 arraysize: 5
 ```
 
+Only single-dimensional arrays are supported.
+
+---------------
+# VALUE RANGE SPECIFICATION
+
+Methods, events, and properties can optionally specify a range of
+legal value that their `in`, `out`, `error` and property can take.
+
+A range is specified as a list of values that are legal for a given
+parameter or property to have.
+
+Range definitions can be used to verify that the contract of a method,
+event, or property is not violated by transmitting illegal values
+between caller and callee, or publisher and subscriber in the case of
+properties.
+
+If `range` is omitted, any value that can be carried by the given
+datatype is a legal value.
+
+A range is specified as a string with a boolean expression consisting
+of relational operators and operands set to the value being verified.
+
+The `$` operand is the value being verified.
+
+String values are double quoted (`"`). Double quotes within a string are expressed as two double quotes (`""`).
+
+
+`$ == 10 or $ == 20`
+
+`$ in (1,3,4,5)`
+
+`$ <= 20`
+
+`$ == "a value"`
+
+`$ == "a ""double-quoted"" value"`
+
+`$ != 1.2 and ($ < 1.0 or $ > 1.4)`
```

Comment:
Float comparison using `!=`, isn't that risky also in VSC. better to use integer values in this example?

Reply:
I think we need to support this, even if we have to resort to a `near()` function that limits precision.

Reply:
We can bring this up in a separate ticket if we are worried about floating point constant comparisons